Newsletters: add 332 (2024-12-06) (#2047)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2024-12-06-newsletter.md

---
title: 'Bitcoin Optech Newsletter #332'
permalink: /en/newsletters/2024/12/06/
name: 2024-12-06-newsletter
slug: 2024-12-06-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Continued discussion about consensus cleanup soft fork proposal:**
  Antoine Poinsot [posted][poinsot time warp] to the existing Delving
  Bitcoin thread about the [consensus cleanup][topic consensus cleanup]
  soft fork proposal.  In addition to the already proposed fix for the
  classic [time warp vulnerability][topic time warp], he proposed also
  including a fix for the recently discovered Zawy-Murch time warp (see
  [Newsletter #316][news316 time warp]).  He favored a fix originally
  proposed by Mark "Murch" Erhardt, "require that the last block in a
  difficulty period _N_ has a higher timestamp than the first block in
  the same difficulty period _N_".

  Anthony Towns [favored][towns time warp] an alternative solution
  proposed by Zawy that would forbid any block from claiming it was
  produced more than two hours before any previous block.  Zawy

>   [noted][zawy time warp] that his every-block solution increased the

Reviewer: ```suggestion
  [noted][zawy time warp] that his every-block solution would increase the
```

release candidates._

- [Core Lightning 24.11rc2][] is a release candidate for the next major
  version of this popular LN implementation.

- [LDK 0.18.4-beta.rc1][] is a release candidate for a minor version of
  this popular LN implementation.

<!-- FIXME:harding to update Thursday -->

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #30708][] adds the `getdescriptoractivity` RPC command that
  retrieves all transactions associated with a [descriptor][topic descriptors]
  within a specified set of blockhashes, allowing wallets to interact with
  Bitcoin Core in a stateless manner. This command is particularly useful when

>   used in conjunction with `scanblocks` (See Newsletter [#222][news222

Reviewer 2: ```suggestion
  used in conjunction with `scanblocks` (see Newsletter [#222][news222
```

  scanblocks]), which identifies the set of blockhashes containing transactions
  associated with a descriptor.

- [Core Lightning #7832][] introduces [anchor outputs][topic anchor outputs] for
  non-urgent unilateral close transactions, starting with a block target of 2016
  blocks (approximately 2 weeks) and gradually reducing to 12 blocks. The
  broadcast timestamp will be tracked to ensure consistent behaviour across

>   restarts. Previously, these transactions did not have anchor outputs by

Author: I think this is incorrect.  Anchor-style channels must have anchor _outputs_ and, since unilateral closes are signed by both parties, no single party can decide not to include them.  What I think this PR does is always create a spend of the local party's anchor output (a child transaction).  That way the user can either RBF the child for higher CPFP fee bumping or the user can CPFP the child (if the carve out limits haven't been hit).

I'm going to make edits to the above effect, but I wanted to leave this note in case I'm wrong.

  default, making it difficult to add one manually, or impossible to use
  [CPFP][topic cpfp] fee bumping if left without an anchor.

> - [LND #8270][] implements the [channel quiescence][topic channel commitment

Reviewer 2: suggestion to remove this first topic channel commitment upgrades reference as we reference it later, in a more applicable spot

  upgrades] protocol as specified in [BOLT2][] (see Newsletter [#309][news309
  quiescence]), which is a prerequisite for [dynamic commitment][topic channel
  commitment upgrades] and [splicing][topic splicing]. The protocol allows a
  node to both respond to a peer's quiescence request and initiate the process
  using new `ChannelUpdateHandler` operations. This PR also adds a configurable
  timeout mechanism to handle unresponsive peers by disconnecting them if the
  quiescent state persists for too long without resolution.

- [LND #8390][] introduces support for setting and relaying an experimental
  [HTLC endorsement][topic htlc endorsement] signaling field in
  `update_add_htlc` messages, aimed at researching [channel jamming
  attack][topic channel jamming attacks] prevention. If a node receives an HTLC
  with the signaling field, it will relay the field as-is; otherwise, it sets a
  default value of zero. This feature is enabled by default but can be disabled.

- [BIPs #1534][] merges BIP349 for the specification of `OP_INTERNALKEY`, a

>   new BIP342 tapscript only opcode that puts the taproot's internal key on the

Reviewer 2: ```suggestion
  new BIP342 [tapscript][topic tapscript]-only opcode that puts the taproot's internal key on the
```

>   new BIP342 tapscript only opcode that puts the taproot's internal key on the

Reviewer 3: ```suggestion
  new BIP342 tapscript only opcode that puts the taproot internal key on the
```

  stack. Authors of scripts need to know the internal key before they can pay to

>   an output, so they could include an internal key directly in a script. See

Reviewer: Just to confirm: is this sentence right? I am a bit lost in it but that may be due to my lack of knowledge.

Author: It's correct but I'll add some clarification.

>   an output, so they could include an internal key directly in a script. See
>   Newsletter [#285][news285 bip349].

Reviewer 2: ```suggestion
  an output, so they could include an internal key directly in a script (see
  Newsletter [#285][news285 bip349]).
```

  indefinitely.  Most attacks on LN will need to delay broadcast for
  between 32 blocks (Core Lightning defaults) and 140 blocks (Eclair
  defaults), which at 10 sats/vbyte would cost between 1.3 BTC ($130,000
  USD) and 5.9 BTC ($590,000 USD), although Riard notes that a careful
  attacker who is well connected to other relay nodes (or directly to
  large miners) may be able to significantly reduce these costs.

  The second version of the attack, called _low overflow_ by Riard,
  takes advantage of Bitcoin Core refusing to allow its queue of
  unrequested transactions to grow beyond 5,000 per peer.  The attacker
  sends the victim a huge number of transactions at a minimum feerate;
  the victim announces these to its honest peers and the peers queue the
  announcements; periodically, they attempt to drain the queue by
  requesting the transactions but a deficit builds until it reaches the
  5,000 announcement limit.  At that point, the peers ignore further
  announcements until the queue has partially drained.  If the victim's
  honest transaction is announced during that time, the peers will
  ignore it.  This attack can be significantly cheaper than the

>   high-overflow variant because it the attacker's junk transactions can

Reviewer: ```suggestion
  high-overflow variant because the attacker's junk transactions can
```

---
title: 'Bitcoin Optech Newsletter #332'
permalink: /en/newsletters/2024/12/06/
name: 2024-12-06-newsletter
slug: 2024-12-06-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Continued discussion about consensus cleanup soft fork proposal:**
  Antoine Poinsot [posted][poinsot time warp] to the existing Delving
  Bitcoin thread about the [consensus cleanup][topic consensus cleanup]
  soft fork proposal.  In addition to the already proposed fix for the
  classic [time warp vulnerability][topic time warp], he proposed also
  including a fix for the recently discovered Zawy-Murch time warp (see
  [Newsletter #316][news316 time warp]).  He favored a fix originally

>   proposed by Mark "Murch" Erhardt, "require that the last block in a

Reviewer 2: ```suggestion
  proposed by Mark "Murch" Erhardt, to "require that the last block in a
```

  difficulty period _N_ has a higher timestamp than the first block in
  the same difficulty period _N_".

  Anthony Towns [favored][towns time warp] an alternative solution
  proposed by Zawy that would forbid any block from claiming it was
  produced more than two hours before any previous block.  Zawy
  [noted][zawy time warp] that his every-block solution increased the
  risk of miners losing money from running outdated software but it

>   would also make timestamps more accurate for other uses, such as time
>   locks (e.g., [HTLCs][topic htlc]).

Reviewer 2: ```suggestion
  would also make timestamps more accurate for other uses, such as
  [timelocks][topic timelocks] (e.g., [HTLCs][topic htlc]).
```

  Separately, both on [Delving Bitcoin][poinsot duptx delv] and the
  [Bitcoin-Dev mailing list][poinsot duptx ml], Poinsot sought feedback
  about which proposed solution to use to prevent block 1,983,702 and
  some later blocks from [duplicating][topic duplicate transactions] a
  previous coinbase transaction (which could lead to loss of money and
  the creation of an attack vector).  Four proposed solutions are
  presented, all of which would only directly affect miners---so

>   feedback from miners would be especially appreciated.

Reviewer 2: Could link "feedback from miners" to https://groups.google.com/g/bitcoinminingdev/c/qyrPzU1WKSI

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

- [Core Lightning 24.11rc2][] is a release candidate for the next major
  version of this popular LN implementation.

> - [LDK 0.18.4-beta.rc1][] is a release candidate for a minor version of

Reviewer 2: ```suggestion
- [LND 0.18.4-beta.rc1][] is a release candidate for a minor version of
```

  this popular LN implementation.

<!-- FIXME:harding to update Thursday -->

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #30708][] rpc: add getdescriptoractivity

- [Core Lightning #7832][] anchors: create low priority anchor to spend commit tx within a week.

- [LND #8270][] DynComms [1/n]: Implement Quiescence Protocol

- [LND #8390][] Add Experimental Endorsement Signalling

- [BIPs #1534][] Add BIP 349: OP_INTERNALKEY

{% include references.md %}
{% include linkers/issues.md v=2 issues="30708,7832,8270,8390,1534" %}
[core lightning 24.11rc2]: https://github.com/ElementsProject/lightning/releases/tag/v24.11rc2

> [ldk 0.18.4-beta.rc1]: https://github.com/lightningnetwork/lnd/releases/tag/v0.18.4-beta.rc1

Reviewer 2: ```suggestion
[lnd 0.18.4-beta.rc1]: https://github.com/lightningnetwork/lnd/releases/tag/v0.18.4-beta.rc1
```

  previous coinbase transaction (which could lead to loss of money and
  the creation of an attack vector).  Four proposed solutions are
  presented, all of which would only directly affect miners---so
  [feedback from miners][mining-dev] would be especially appreciated.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

- [Eclair v0.11.0][] is the latest release of this popular LN node
  implementation.  It "adds official support for BOLT12 [offers][topic
  offers] and makes progress on liquidity management features

>   ([splicing][topic splicing], [liquidity advs][topic liquidity

Reviewer: ```suggestion
  ([splicing][topic splicing], [liquidity ads][topic liquidity
```
